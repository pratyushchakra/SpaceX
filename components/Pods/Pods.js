import { withContext } from '../../utils/Context';
import { Fragment, useEffect } from 'react';
import PropTypes from 'prop-types';

import styles from './Pods.style'
import { fetchPodsData } from '../../utils/utils';

const Pod = ({ launch }) => (
    <div className="pod">
        <div className="image-container">
            <img src={launch.links.mission_patch} width="200" />
        </div>
        <p className="mission-name">{launch.mission_name} #{launch.flight_number}</p>
        <div className="text-holder"><p className="bold-text">Mission IDs:</p><p>{launch.mission_id}</p></div>
        <div className="text-holder"><p className="bold-text">Launch Year:</p><p>{launch.launch_year}</p></div>
        <div className="text-holder"><p className="bold-text">Successful Launch:</p><p>{launch.launch_success ? 'True' : 'False'}</p></div>
        <div className="text-holder"><p className="bold-text">Successful Landing:</p><p>{launch.isLandingTrue ? 'True' : 'False'}</p></div>
        <style jsx>{styles}</style>
    </div>
)
const Pods = ({ store }) => {
    if (!store) return null;

    const { launchConfig, filterConfig } = store;


    const { allLaunches, setLaunchData } = launchConfig;
    const { filterData } = filterConfig;
    useEffect(() => {
        if (!filterData || Object.entries(filterData).length === 0) return
        fetchPodsData(filterData, setLaunchData);
    }, [filterData]);


    return (
        <Fragment>
            {
                allLaunches.map(launch => (
                    <Pod key={launch.flight_number} launch={launch} />
                ))
            }
            <style jsx>{styles}</style>
        </Fragment>
    )
}
Pods.propTypes = {
    store: PropTypes.object.isRequired
}
export { Pods };
export default withContext(Pods);