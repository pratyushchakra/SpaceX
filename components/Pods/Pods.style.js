import css from 'styled-jsx/css';

const PodStyles = css`
.pod {
    text-align: center;
    background: #fff;
    padding: 15px;
    margin: 0 15px 15px 0;
    font-size:14px;
}
img {
    width: 100%;
}
.text-holder {
    text-align: left;
    display: flex;
    margin-bottom: 6px;
}
.bold-text {
    font-weight: bold;
    margin-right: 5px;
}
.image-container {
    background: #eee;
    margin: 0 auto;
}
.mission-name {
    text-align: left;
    font-weight: bold;
    color: #485499;
    margin: 10px 0;
    font-size: 15px;
}
@media only screen and (max-width: 700px) {
.pod {
    margin: 0 0 15px 0;
}
}
`
export default PodStyles;