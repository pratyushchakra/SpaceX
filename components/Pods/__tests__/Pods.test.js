import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Pods } from '../Pods';
import mock from '../../../utils/mock';
Enzyme.configure({ adapter: new Adapter() });

let store
beforeEach(() => {
    store = {
        "launchConfig": {
            "allLaunches": mock,
            "setLaunchData": jest.fn()
        },
        "filterConfig": {
            "filterData": {},
            "setFilterData": jest.fn()
        }
    }
})
describe('Pods Component', () => {
    it('renders Pods component with proper data', () => {
        const wrapper = shallow(
            <Pods store={store} />
        );
        expect(wrapper.debug()).toMatchSnapshot();
    });
    it('does not render Pod component if there is not launch data', () => {
        store.launchConfig.allLaunches = []
        const wrapper = shallow(
            <Pods store={store} />
        );
        expect(wrapper.debug()).toMatchSnapshot();
        expect(wrapper.find('Pod').length).toBe(0)
    })
});