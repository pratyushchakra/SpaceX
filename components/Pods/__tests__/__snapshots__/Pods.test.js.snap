// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pods Component does not render Pod component if there is not launch data 1`] = `
"<Fragment>
  <JSXStyle id=\\"1350581516\\">
    .
    p
    o
    d
    .
    j
    s
    x
    -
    1
    3
    5
    0
    5
    8
    1
    5
    1
    6
    {
    t
    e
    x
    t
    -
    a
    l
    i
    g
    n
    :
    c
    e
    n
    t
    e
    r
    ;
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    #
    f
    f
    f
    ;
    p
    a
    d
    d
    i
    n
    g
    :
    1
    5
    p
    x
    ;
    m
    a
    r
    g
    i
    n
    :
    0
     
    1
    5
    p
    x
     
    1
    5
    p
    x
     
    0
    ;
    f
    o
    n
    t
    -
    s
    i
    z
    e
    :
    1
    4
    p
    x
    ;
    }
    i
    m
    g
    .
    j
    s
    x
    -
    1
    3
    5
    0
    5
    8
    1
    5
    1
    6
    {
    w
    i
    d
    t
    h
    :
    1
    0
    0
    %
    ;
    }
    .
    t
    e
    x
    t
    -
    h
    o
    l
    d
    e
    r
    .
    j
    s
    x
    -
    1
    3
    5
    0
    5
    8
    1
    5
    1
    6
    {
    t
    e
    x
    t
    -
    a
    l
    i
    g
    n
    :
    l
    e
    f
    t
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    f
    l
    e
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    m
    s
    -
    f
    l
    e
    x
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    f
    l
    e
    x
    ;
    m
    a
    r
    g
    i
    n
    -
    b
    o
    t
    t
    o
    m
    :
    6
    p
    x
    ;
    }
    .
    b
    o
    l
    d
    -
    t
    e
    x
    t
    .
    j
    s
    x
    -
    1
    3
    5
    0
    5
    8
    1
    5
    1
    6
    {
    f
    o
    n
    t
    -
    w
    e
    i
    g
    h
    t
    :
    b
    o
    l
    d
    ;
    m
    a
    r
    g
    i
    n
    -
    r
    i
    g
    h
    t
    :
    5
    p
    x
    ;
    }
    .
    i
    m
    a
    g
    e
    -
    c
    o
    n
    t
    a
    i
    n
    e
    r
    .
    j
    s
    x
    -
    1
    3
    5
    0
    5
    8
    1
    5
    1
    6
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    #
    e
    e
    e
    ;
    m
    a
    r
    g
    i
    n
    :
    0
     
    a
    u
    t
    o
    ;
    }
    .
    m
    i
    s
    s
    i
    o
    n
    -
    n
    a
    m
    e
    .
    j
    s
    x
    -
    1
    3
    5
    0
    5
    8
    1
    5
    1
    6
    {
    t
    e
    x
    t
    -
    a
    l
    i
    g
    n
    :
    l
    e
    f
    t
    ;
    f
    o
    n
    t
    -
    w
    e
    i
    g
    h
    t
    :
    b
    o
    l
    d
    ;
    c
    o
    l
    o
    r
    :
    #
    4
    8
    5
    4
    9
    9
    ;
    m
    a
    r
    g
    i
    n
    :
    1
    0
    p
    x
     
    0
    ;
    f
    o
    n
    t
    -
    s
    i
    z
    e
    :
    1
    5
    p
    x
    ;
    }
    @
    m
    e
    d
    i
    a
     
    o
    n
    l
    y
     
    s
    c
    r
    e
    e
    n
     
    a
    n
    d
     
    (
    m
    a
    x
    -
    w
    i
    d
    t
    h
    :
    7
    0
    0
    p
    x
    )
    {
    .
    p
    o
    d
    .
    j
    s
    x
    -
    1
    3
    5
    0
    5
    8
    1
    5
    1
    6
    {
    m
    a
    r
    g
    i
    n
    :
    0
     
    0
     
    1
    5
    p
    x
     
    0
    ;
    }
    }
  </JSXStyle>
</Fragment>"
`;

exports[`Pods Component renders Pods component with proper data 1`] = `
"<Fragment>
  <Pod launch={{...}} />
  <Pod launch={{...}} />
  <JSXStyle id=\\"1350581516\\">
    .
    p
    o
    d
    .
    j
    s
    x
    -
    1
    3
    5
    0
    5
    8
    1
    5
    1
    6
    {
    t
    e
    x
    t
    -
    a
    l
    i
    g
    n
    :
    c
    e
    n
    t
    e
    r
    ;
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    #
    f
    f
    f
    ;
    p
    a
    d
    d
    i
    n
    g
    :
    1
    5
    p
    x
    ;
    m
    a
    r
    g
    i
    n
    :
    0
     
    1
    5
    p
    x
     
    1
    5
    p
    x
     
    0
    ;
    f
    o
    n
    t
    -
    s
    i
    z
    e
    :
    1
    4
    p
    x
    ;
    }
    i
    m
    g
    .
    j
    s
    x
    -
    1
    3
    5
    0
    5
    8
    1
    5
    1
    6
    {
    w
    i
    d
    t
    h
    :
    1
    0
    0
    %
    ;
    }
    .
    t
    e
    x
    t
    -
    h
    o
    l
    d
    e
    r
    .
    j
    s
    x
    -
    1
    3
    5
    0
    5
    8
    1
    5
    1
    6
    {
    t
    e
    x
    t
    -
    a
    l
    i
    g
    n
    :
    l
    e
    f
    t
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    f
    l
    e
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    m
    s
    -
    f
    l
    e
    x
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    f
    l
    e
    x
    ;
    m
    a
    r
    g
    i
    n
    -
    b
    o
    t
    t
    o
    m
    :
    6
    p
    x
    ;
    }
    .
    b
    o
    l
    d
    -
    t
    e
    x
    t
    .
    j
    s
    x
    -
    1
    3
    5
    0
    5
    8
    1
    5
    1
    6
    {
    f
    o
    n
    t
    -
    w
    e
    i
    g
    h
    t
    :
    b
    o
    l
    d
    ;
    m
    a
    r
    g
    i
    n
    -
    r
    i
    g
    h
    t
    :
    5
    p
    x
    ;
    }
    .
    i
    m
    a
    g
    e
    -
    c
    o
    n
    t
    a
    i
    n
    e
    r
    .
    j
    s
    x
    -
    1
    3
    5
    0
    5
    8
    1
    5
    1
    6
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    #
    e
    e
    e
    ;
    m
    a
    r
    g
    i
    n
    :
    0
     
    a
    u
    t
    o
    ;
    }
    .
    m
    i
    s
    s
    i
    o
    n
    -
    n
    a
    m
    e
    .
    j
    s
    x
    -
    1
    3
    5
    0
    5
    8
    1
    5
    1
    6
    {
    t
    e
    x
    t
    -
    a
    l
    i
    g
    n
    :
    l
    e
    f
    t
    ;
    f
    o
    n
    t
    -
    w
    e
    i
    g
    h
    t
    :
    b
    o
    l
    d
    ;
    c
    o
    l
    o
    r
    :
    #
    4
    8
    5
    4
    9
    9
    ;
    m
    a
    r
    g
    i
    n
    :
    1
    0
    p
    x
     
    0
    ;
    f
    o
    n
    t
    -
    s
    i
    z
    e
    :
    1
    5
    p
    x
    ;
    }
    @
    m
    e
    d
    i
    a
     
    o
    n
    l
    y
     
    s
    c
    r
    e
    e
    n
     
    a
    n
    d
     
    (
    m
    a
    x
    -
    w
    i
    d
    t
    h
    :
    7
    0
    0
    p
    x
    )
    {
    .
    p
    o
    d
    .
    j
    s
    x
    -
    1
    3
    5
    0
    5
    8
    1
    5
    1
    6
    {
    m
    a
    r
    g
    i
    n
    :
    0
     
    0
     
    1
    5
    p
    x
     
    0
    ;
    }
    }
  </JSXStyle>
</Fragment>"
`;
